dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/icon/IconController.java

```diff
@@ -137,6 +140,25 @@ public void getIconData(@PathVariable String key, HttpServletResponse response)
     return icons.stream().map(iconMapper::from).toList();
   }
 
+  @GetMapping(value = "/type/{type}")
```

Comment:
I am just curious about the API design here, why type is not a RequestParam?
Default value of type param could be both so in case is not specified it will have the same behavior of `getAllIcons`, that could be removed then

Reply:
right. Initially, i created `api/icons?type` and that was conflicting with `api/icons?iconkey`. So then i had to introduce another path `api/icons/type`. but yes `type` can be RequestParam.

Reply:
What do you mean they were conflicting?

In general, it would be good to have one flow for retrieving and have all the parameters in the criteria.
Here we already have 3 different calls to the service (getIcons(), getIcons(keywords) and getIcons(criteria)) that are doing the same thing but filtering on different fields.
We could have just one getIcons(criteria) with all the params and the flow would be much simpler

Reply:
Ambiguous mapping `IconController#getIconByType(String, String[])` and `IconController#getAllIcons(String[])`

Reply:
for multiple getIcon methods. I will check if they all can be replaced by one.

Reply:
having everything in one getIcon(criteria) would make it very messy. It looks abit redundant now, but i think
in the current shape they are abit simpler.
Custom types are coming from database where as defaults are bundled in the war file. In some cases we need all of them but in others, we need all but with keyword filtration. So there are many scenarios that a single method needs to cover and that would make one method unnecessarily complex.